Fixed some bugs and added some features

kaminomisosiru opened the pull request:
### Bug fix

- Fix some typo.
- Changed to display some text in the language appropriate to the system language.
- Fixed an issue where moving files from the left panel to the right panel files are not displayed in the panel.
- Directory is loaded recursively so that the directory of arbitrary depth can be loaded.

### add new features

- When taking a screen shot of the analysis result, display a dialog for selecting a place to save.
- The program was terminated when the "Clear" button in the main window was pressed

Commit:
Fixed the problem about panel

Fixed the problem that the selected file is not displayed on the right-side panel after clicking the button `>>`

Commit:
Fixed the problem about language

Because the tooltip of the '?' Button is in Spanish, changed it to English.

Commit:
Bug fixed

Fix the issue where Token counting test does not run.
Change the delimiter of argument of StringTokenizer method.

Commit:
Fixed typo

Commit:
Fix the problem about file loading in Test window

Fixed to correctly analyze files in the loaded directory.

Commit:
Implement Exit button

Change Cancel button to Exit button.
Change related function or variable name and display text message.

Commit:
Multilingualize text on gui window

Commit:
Load submission directory recursively

Commit:
Implement dialog to choose where to save screenshots

Commit:
Add dialog when a screenshot file has been saved

Commit:
Add error dialog when a screenshot file has not been saved

Commit:
Fixed throwing NullPointerException when saving result file

Commit:
Fix incorrect dialog display

Commit:
Change display text in AC window

Commit:
Add extension to screenshot file automatically

Commit:
Skip hidden file when loading a source file.

Commit:
Change screenshot save button text and add tooltip on the button

Commit:
Update license text

Commit:
Add tooltip on Clear button

Commit:
Update language property file

Commit:
Change testkey in TokenCountingTest

manuel-freire commented:
Wow, looks nice - great work. I will need some time to review it before merging, though.

manuel-freire commented on ac-core/src/main/java/es/ucm/fdi/ac/test/TokenCountTest.java:
```diff
@@ -64,7 +64,7 @@ public void preprocess(Submission s) {
 
 		TreeMap<Integer, Integer> counter = new TreeMap<Integer, Integer>();
-		StringTokenizer st = new StringTokenizer(tokens, "\n\r");
+		StringTokenizer st = new StringTokenizer(tokens, " ");
```
typical antlr4 tokenized strings look like `3a 13 9 36 1r 3b 1s -1` (for the `sample.java` file included in ac-lexers test), so yes, this works. The old tokenizer (from PMD) generated output in columns. I would, however, prefer to use more whitespace characters to future-proof things somewhat: " \n\r\t"

manuel-freire commented on ac-core/src/main/java/es/ucm/fdi/ac/Analysis.java:
```diff
@@ -150,6 +134,34 @@ public void loadSources(SourceSet sources) throws IOException {
+			HashMap<String, Submission> unique, int i) {
+		if (tn.getFile().isDirectory()) {
+			for (FileTreeNode fn : tn.getLeafChildren()) {
+				addSubmissionRecursive(fn, unique, i);
```
if `i` is incremented inside recursive calls, these increments will be invisible once the calls return. This results in duplicate ids being assigned, which is bad. Given that we have the `unique` hashmap, we can dispose with i, and just use the hashmap's current size as unique identifiers. In any case, as it is, this code is buggy.

Commit:
Change the delimiter for StringTokenizer method in TokenCountTest.java

Use \r, \n, \t as delimiter

Commit:
Change submission counter

Count the number of submissions using a static class variable.

kaminomisosiru commented:
Thank you for your review.
I have fixed the improvements you pointed out.

- In addition to blanks, I added delimiters "\n\r\t".
- The number of submissions is counted by a static variable.
